packages/react/src/components/DataTable/__tests__/TableSelectRow-test.js

```diff
@@ -23,45 +23,49 @@ describe('DataTable.TableSelectRow', () => {
     };
   });
 
-  it('should render', () => {
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    expect(wrapper).toMatchSnapshot();
-  });
+  describe('renders as expected - Component API', () => {
+    it('should render', () => {
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      expect(container).toMatchSnapshot();
+    });
 
-  it('should render with the provided class name', () => {
-    const customClassName = 'custom-table-select-row-classname';
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} className={customClassName} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    const elements = wrapper.find(`td.${customClassName}`);
-    expect(elements.length).toBe(1);
+    it('should render with the provided class name', () => {
+      const customClassName = 'custom-table-select-row-classname';
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} className={customClassName} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      const elements = container.querySelectorAll(`td.${customClassName}`);
+      expect(elements.length).toBe(1);
```

Comment:
Two things:

1. Instead of using querySelector, use `screen` coupled with a [query](https://testing-library.com/docs/queries/about#using-queries).
2. All the [jest-dom matchers](https://github.com/testing-library/jest-dom#tohaveclass) are available and can make things more readable

There's [a recommended priority](https://testing-library.com/docs/queries/about#priority) for the queries to skew towards using more accessible ones. I think this is one of the few cases where `getByTestId` actually makes sense, despite it being at the bottom of that list.

This test could look something like

```
      render(
        <Table>
          <TableHead>
            <TableRow>
              <TableSelectRow {...mockProps} data-testid="test-row" />
            </TableRow>
          </TableHead>
        </Table>
      );
      expect(screen.getByTestId('test-row').firstChild).toHaveClass(className);
```

Reply:
done! thanks so much for the RTL references 🙏🏻

Reply:
I ended up going with `expect(screen.getByRole('row').firstChild).toHaveClass(mockProps.className);` btw because `TableSelectRow` is not spreading props so it's not passing off the datatestId prop